runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsOverlay.java

```diff
@@ -0,0 +1,215 @@
+import net.runelite.api.WallObject;
+import net.runelite.api.queries.WallObjectQuery;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.util.QueryRunner;
+
+import javax.inject.Inject;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.util.List;
+import java.util.stream.IntStream;
+
+class BarrowsOverlay extends Overlay
+{
+	private final Client client;
+	private final BarrowsConfig config;
+
+	private static final int REGION_SIZE = 104;
+	private static final int MAX_DISTANCE = 2350;
+
+	private static final int[] barrowsWalls =
+	{
+		20678, 20681, 20682, 20683, 20684, 20685, 20686, 20687,
+		20688, 20689, 20690, 20691, 20692, 20693, 20694, 20695,
+		20696, 20697, 20700, 20701, 20702, 20703, 20704, 20705,
+		20706, 20707, 20708, 20709, 20710, 20711, 20712, 20713,
+		20714, 20715, 20728, 20730
+	};
+
+	private static final int[] barrowsLadders = { 20675, 20676, 20677 };
+
+	@Inject
+	QueryRunner queryRunner;
+
+	@Inject
+	BarrowsOverlay(Client client, BarrowsConfig config)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+
+		this.client = client;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics, Point parent)
+	{
+		if (!config.enabled())
+		{
+			return null;
+		}
+
+		Player local = client.getLocalPlayer();
+
+		//Determine if you are in the tunnels based off if a barrows wall is present
+		if (inTunnels(local) && config.showMinimap())
+		{
+			//NPC yellow dot
+			List<NPC> npcs = client.getNpcs();
+			for (NPC npc : npcs)
+			{
+				net.runelite.api.Point minimapLocation = npc.getMinimapLocation();
+				if (minimapLocation != null)
+				{
+					graphics.setColor(Color.yellow);
+					graphics.fillOval(minimapLocation.getX(), minimapLocation.getY(), 4, 4);
+				}
+			}
+
+			//Render barrows walls/doors
+			renderObjects(graphics, local);
+
+			//Player white dot
+			graphics.setColor(Color.white);
+			graphics.fillRect(local.getMinimapLocation().getX(), local.getMinimapLocation().getY(), 3, 3);
+		}
+		else if (config.showBrotherLoc())
+		{
+				renderBarrowsBrothers(graphics);
+		}
+
+		return null;
+	}
+
+	private void renderObjects(Graphics2D graphics, Player local)
+	{
+		Region region = client.getRegion();
+		Tile[][][] tiles = region.getTiles();
+
+		int z = client.getPlane();
+
+		for (int x = 0; x < REGION_SIZE; ++x)
+		{
+			for (int y = 0; y < REGION_SIZE; ++y)
+			{
+				Tile tile = tiles[z][x][y];
+
+				if (tile == null)
+				{
+					continue;
+				}
+				if (local.getLocalLocation().distanceTo(tile.getLocalLocation()) <= MAX_DISTANCE)
+				{
+					renderBarrowsWalls(graphics, tile);
+				}
+			}
+		}
+	}
+
+	private void renderBarrowsWalls(Graphics2D graphics, Tile tile)
+	{
+		//Drawing walls/doors
+		WallObject wallObject = tile.getWallObject();
+		if (wallObject != null && IntStream.of(barrowsWalls).anyMatch(x -> x == wallObject.getId()))
```

Comment:
list#contains (and creating barrowsWalls as list) would be a lot faster than creating stream on the fly and using anyMatch on it.